src/app/features/reporting/reporting.service.ts

```diff
@@ -0,0 +1,172 @@
+import { Injectable } from "@angular/core";
+import { QueryService } from "./query.service";
+
+export interface Aggregation {
+  query: string;
+  groupBy?: string[];
+  label?: string;
+  aggregations?: Aggregation[];
+}
+
+export interface ReportRow {
+  header: { label: string; values: any[]; result: number };
+  subRows: ReportRow[];
+}
+
+@Injectable({
+  providedIn: "root",
+})
+export class ReportingService {
+  private aggregations: Aggregation[] = [];
+  private fromDate: Date;
+  private toDate: Date;
+
+  constructor(private queryService: QueryService) {}
+
+  public setAggregations(aggregations: Aggregation[]) {
+    this.aggregations = aggregations;
+  }
+
+  public calculateReport(from?: Date, to?: Date): Promise<ReportRow[]> {
```

Comment:
any benefit to persist the date range in the service rather than only directly pass it directly into calculation method?